lib/api/controllers/securityController.js

```diff
@@ -697,13 +720,21 @@ class SecurityController extends BaseController {
    * @param {Request} request
    * @returns {Promise<Object>}
    */
-  deleteUser(request) {
-    assertHasId(request);
+  async deleteUser(request) {
+    const
+      userId = this.getId(request),
+      options = { refresh: request.input.args.refresh };
 
-    const options = { refresh: request.input.args.refresh };
+    const user = await this.kuzzle.repositories.user.load(userId);
 
-    return this.kuzzle.repositories.user.load(request.input.resource._id)
-      .then(user => this.kuzzle.repositories.user.delete(user, options));
+    await this.kuzzle.repositories.user.delete(user, options);
+
+    // @todo test this
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)